File: src-tauri/src/devtools.rs
// DevTools 开关（08 T7；View 菜单装配后续归 12 窗口外壳）
//
// 编译门控：tauri 的 open/close/is_devtools_open 方法带
// #[cfg(any(debug_assertions, feature = "devtools"))]——debug 构建默认可用，
// release 须经本 crate 的 devtools feature（映射 tauri/devtools）显式启用；
// 未启用构建下命令退化为无操作返回 false（前端快捷键不报错）。

/// 切换 DevTools 开合（AC-T7-1；Shift+F12 由前端窗口级快捷键触发本命令）
///
/// @returns 切换后状态：true=已打开，false=已关闭（或当前构建不支持 DevTools）
#[tauri::command]
pub fn toggle_devtools(window: tauri::WebviewWindow) -> Result<bool, String> {
    #[cfg(any(debug_assertions, feature = "devtools"))]
    {
        if window.is_devtools_open() {
            window.close_devtools();
            Ok(false)
        } else {
            window.open_devtools();
            Ok(true)
        }
    }
    #[cfg(not(any(debug_assertions, feature = "devtools")))]
    {
        // 未启用构建：显式消费参数避免 unused 警告；无操作返回 false
        let _ = window;
        Ok(false)
    }
}

File: src/features/theme/devtools.ts
// DevTools 快捷键（08 T7）：Shift+F12 窗口级组合（编辑器内快捷键归 01 keymap 注册表，
// 本层只处理窗口级组合——app-shortcuts.ts 同一分层）。命令本体见 Rust devtools.rs，
// debug 构建默认可用、release 需 devtools feature（不支持时命令返回 false 不报错）。
import { toggleDevtools } from "../../services/theme-io";

/**
 * 注册 Shift+F12 窗口级快捷键（View → Toggle DevTools；12 窗口外壳可整体接管）
 * @returns 注销函数（App 卸载调用）
 */
export function registerDevtoolsShortcut(): () => void {
  const onKeydown = (event: KeyboardEvent): void => {
    // 编辑器 keymap 已消费的按键不得重复触发（TASK 10#3 清单外同型缺口收口，
    // settings-shortcuts.ts 同款先例：prosemirror-view 命中仅 preventDefault 不阻断传播）
    if (event.defaultPrevented) return;
    // 仅 Shift+F12 命中；无 Shift 的 F12 留给浏览器默认行为
    if (!(event.shiftKey && event.key === "F12")) return;
    event.preventDefault();
    // 开合结果无需消费；失败（构建不支持/IPC 异常）仅记录不打断交互
    void toggleDevtools().catch((e: unknown) => {
      console.error("[MarkWell] DevTools 切换失败", e);
    });
  };
  window.addEventListener("keydown", onKeydown);
  return () => window.removeEventListener("keydown", onKeydown);
}
